torchmetrics/image/map.py

```diff
@@ -303,21 +303,21 @@ def _get_coco_format(self, is_pred: bool = False) -> Dict:
         annotations = []
         annotation_id = 1  # has to start with 1, otherwise COCOEval results are wrong
 
-        for i, _ in enumerate(self.boxes):
+        for i, _ in enumerate(self.groundtruth_boxes):
```

Comment:
this assumes that `len(self.groundtruth_boxes) == len(self.detection_boxes)`, which is quite unlikely to happen.

Suggestion:

```
iterable =  self.detection_boxes if is_pred else self.groundtruth_boxes
for i, _ in enumerate(iterable):
```

Alternatively, you could also jointly iterate over the corresponding variables directly by zipping them.

Reply:
IMHO it should always be the same length, because its a `List[List[boxes]]`, first list length is the amount of images, which is checked in `_input_validator (if len(preds) != len(targets))`. 

Each iteration represents the results for a single sample.

I'd prefer the zip, but that does not work with `scores = None` for groundtruth data.

Reply:
Ahh okay, then I misread/misunderstood the API of the metric, thanks for clarifying !

Reply:
No problem, thanks for double checking!